src/read/stream.rs

```diff
@@ -85,6 +85,13 @@ pub async fn read_metadata<R: AsyncRead + AsyncSeek + Send + std::marker::Unpin>
         Cursor::new(buffer)
     };
 
+    parse_metadata(reader).await
+}
+
+/// Parse loaded metadata bytes
+pub async fn parse_metadata(
+    reader: Cursor<Vec<u8>>,
+) -> Result<FileMetaData> {
     let mut prot = TCompactInputProtocol::new(reader);
```

Comment:
This code is equal to what we have in `read::metadata::read_metadata`. Maybe we could declare this function in `crate::metadata`, since it is not IO-bounded, rename it to `deserialize_metadata` (to be consistent with other parts of the crate), and reuse it at the end of `read::metadata::read_metadata`?